packages/server/modules/core/domain/users/operations.ts

```diff
@@ -256,3 +256,12 @@ export type AdminGetInviteList = (args: CollectionQueryArgs) => Promise<{
   totalCount: number
   items: ServerInviteGraphQLReturnType[]
 }>
+
+export type SetUserActiveWorkspace = (args: {
+  userId: string
+  workspaceSlug: string | null
+}) => Promise<void>
+export type SetUserActiveProject = (args: {
+  userId: string
+  projectId: string | null
```

Comment:
For projects we don't need the ID, it's just "you are in the projects world", for reference this is how the state is atm in the FE. So maybe this could just be a boolean?

```
  useState<{
    activeWorkspaceSlug: string | null
    isProjectsActive: boolean
  }>('navigation-state', () => ({
    activeWorkspaceSlug: null,
    isProjectsActive: false
  }))
  ```

Reply:
Let's catch up on this tomorrow :) The problem is that you can be in the "legacy projects world" without being in a specific legacy project. Not really sure what kind of new concept this introduces? We do something similar for the workspace wizard, where we have a JSON object for the state that the FE "controls". But, I think what you are proposing will work